python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -220,10 +232,25 @@ def _DataFrame__dir__(self):
         "_constructor": _FastSlowAttribute("_constructor"),
         "_constructor_sliced": _FastSlowAttribute("_constructor_sliced"),
         "_accessors": set(),
+        "_ipython_canary_method_should_not_exist_": custom_func,
     },
 )
 
 
+def custom_repr_html(obj):
+    # This custom method is need to register a html format
+    # for ipython
+    return _fast_slow_function_call(
+        lambda obj: obj._repr_html_(),
+        obj,
+    )[0]
+
+
+if ipython_shell:
```

Comment:
Do we need to `del ipython_shell` if we get an actual ipython instance?

Reply:
Shouldn't gc be able to clean this properly after program exit? We want this to stay alive for the lifetime of the program.

Reply:
Ah ok sounds good. I just didn't know if we needed any special cleanup if we reference the ipython shell